src/Types/Tuple.h

```diff
@@ -0,0 +1,336 @@
+//
+// Class Vector
+//   Vector class used for vector fields and particle attributes like the coordinate.
+//
+#ifndef IPPL_Tuple_H
+#define IPPL_Tuple_H
+#include <Kokkos_Macros.hpp>
+#include <concepts>
+#include <cstddef>
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+namespace ippl {
+    /*!
+     * @file Tuple.h
+     */
+
+    /*!
+     * @struct TupleImpl
+     * @brief Implementation details for the Tuple class.
+     * @tparam i Current index in the Tuple.
+     * @tparam N Total number of elements in the Tuple.
+     * @tparam Ts Types of elements in the Tuple.
+     */
+    template <std::size_t i, std::size_t N, typename... Ts>
```

Comment:
What do you gain from `N` being its own parameter? Isn't `N == sizeof...(Ts)` invariant?

Reply:
Ts... changes, although I think it can be done better.

Reply:
This should be addressed in the doxygen comment, i.e. indicate that `Ts...` is the types of the remaining elements and not all the types in the tuple.

Reply:
Added a comment